README example needs `iter_mut`

The example in the readme does not compile:

```
// Iterate through all entities that match the query in the world
for (mut pos, vel) in query.iter(&mut world) {
    pos.x += vel.dx;
    pos.y += vel.dy;
}
```

Here is the compile error:

```
error[E0277]: the trait bound `legion_core::query::Write<Position>: legion_core::query::ReadOnly` is not satisfied
  --> src/main.rs:38:33
   |
38 |     for (mut pos, vel) in query.iter(&mut world) {
   |                                 ^^^^ the trait `legion_core::query::ReadOnly` is not implemented for `legion_core::query::Write<Position>`
   |
   = note: required because of the requirements on the impl of `legion_core::query::ReadOnly` for `(legion_core::query::Write<Position>, legion_core::query::Read<Velocity>)`

error: aborting due to previous error
```

I believe `query.iter(...)` needs to be changed to `query.iter_mut(...)`

--- a/readme.md
+++ b/readme.md
@@ -63,7 +63,7 @@ let entities: &[Entity] = world.insert(
 let query = <(Write<Position>, Read<Velocity>)>::query();
 
 // Iterate through all entities that match the query in the world
-for (mut pos, vel) in query.iter(&mut world) {
+for (mut pos, vel) in query.iter_mut(&mut world) {
     pos.x += vel.dx;
     pos.y += vel.dy;
 }

